opencsp/common/lib/cv/spot_analysis/ImagesIterable.py

```diff
@@ -75,12 +75,19 @@ class ImagesIterable(Iterable[CacheableImage]):
 
     def __init__(self, stream: Callable[[int], CacheableImage] | list[str | CacheableImage] | vh.VideoHandler):
         """
+        Initializes the ImagesIterable with the provided stream.
+
         Parameters
         ----------
-        stream : Callable[[int],CacheableImage] | list[str|CacheableImage] | vh.VideoHandler
-            The stream to iterate over. If a callable, then will be passed the
-            current iteration index as an argument.
+        stream : Callable[[int], CacheableImage] | list[str | CacheableImage] | vh.VideoHandler
+            The stream of images to iterate over.
+
+        Raises
+        ------
+        TypeError
+            If the provided stream is not of an expected type (iterator, callable, or list).
```

Comment:
Removing the supported types here, to reduce future upkeep.

```suggestion
            If the provided stream is not one of the supported types.
```